web-components/input/text/text.svelte

```diff
@@ -0,0 +1,117 @@
+
+  interface CommonProps {
+    type?: Props.TextInputType;
+  }
+
+  type TextInputProps = CommonProps & Partial<svelte.JSX.HTMLAttributes<HTMLElementTagNameMap['input']>> & {
+    isRequired?: boolean;
+    isDisabled?: boolean;
+  }
+
+  type $$Props = TextInputProps
+
+  export let value: string = '';
+  export let name: string = '';
+  export let label: string = '';
+  export let error: string = '';
+  export let type: Props.TextInputType = 'text';
+  export let isRequired: boolean = false;
+  export let isDisabled: boolean = false;
+</script>
+
+<div class="form-control" class:required={isRequired} class:error={error} class:disabled={isDisabled}>
+  {#if label}
+    <label for={name}>{label}</label>
+  {/if}
+  <input {...{type}} id={name} {name} bind:value={value} on:input on:change on:focus on:blur required={isRequired ?? undefined} disabled={isDisabled ?? undefined} {...$$restProps} />
+  {#if error}
+    <p class="message">{error}</p>
+  {/if}
+</div>
+
+<style lang="scss">
+  .form-control {
+    --label-text-color: var(--color-text-primary);
+
+    display: flex;
+    flex-direction: column;
+
+    label {
+      font: var(--font-text-small-semibold);
+      padding-bottom: 8px;
+      color: var(--label-text-color);
+    }
+
+    &.required label::after {
+      content: '*';
+      font: var(--font-text-small-regular);
+      color: var(--color-systemfeedback-error-icon);
+      padding-left: 4px;
+    }
+
+    &.error input {
+      --bg: var(--color-systemfeedback-error-background);
+    }
+
+    &.error .message {
+      margin: 0;
+      padding: 0;
+      padding-top: 8px;
+      font: var(--font-text-small-regular);
+      color: var(--color-systemfeedback-error-icon);
+    }
+
+    &.disabled {
+      --label-text-color: var(--color-text-secondary);
+    }
+  }
+
+  input {
+    --text-color: var(--color-text-primary);
+    --placeholder-text-color: var(--color-text-secondary);
+    --bg: var(--color-container-highlight);
+    --radius: var(--radius-8);
+    --padding-y: 10px;
+    --padding-x: 8px;
+    --border-color: transparent;
+    --box-shadow: none;
+
+    all: unset;
+    font: var(--font-text-default-regular);
+    color: var(--text-color);
+    background: var(--bg);
+    border-radius: var(--radius);
+    padding: var(--padding-y) var(--padding-x);
+    border: 1px solid var(--border-color);
+    box-shadow: var(--box-shadow);
+
+    &::placeholder {
+      color: var(--placeholder-text-color);
+    }
+
+    &:hover {
+      --border-color: var(--color-divider-subtle);
+      --box-shadow: var(--effect-elevation-02);
+    }
+
+    &:focus-visible {
+      outline: none;
+      --box-shadow: var(--effect-elevation-focus-state);
+    }
+
+    &:disabled {
+      --text-color: var(--color-text-disabled);
+      --placeholder-text-color: var(--color-text-disabled);
+      --bg: var(--color-container-disabled);
+    }
+  }
+
+  // TODO: is this the best approach?
+  :global(.form-control) + :global(.form-control) {
```

Comment:
Perhaps we need one of the following:
- an independent `.form-item` class that can be imported by this component and any others (including "third party" custom items)
- a separate `form-item.svelte` component that we wrap this and other form elements with 
- Let the consumer deal with spacing by adding their own class names or wrapping flex / grid parents
- Leo provides a FormItemGroup component that provides a default flex parent with defined gaps.

I would also suggest that this should be margin and not padding, unless I'm otherwise mistaken.